[2022-02-16 08:41:22,013] INFO Expiring session 0x10000325aa60001, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 08:41:22,085] WARN Client session timed out, have not heard from server in 43453278ms for session id 0x10000325aa60001 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:22,901] WARN Session 0x10000325aa60001 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 43453278ms for session id 0x10000325aa60001
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-16 08:41:24,101] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,441] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,451] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:61810, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,753] INFO Invalid session 0x10000325aa60001 for client /[0:0:0:0:0:0:0:1]:61810, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 08:41:24,757] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60001 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,757] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 08:41:24,757] INFO EventThread shut down for session: 0x10000325aa60001 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,761] WARN Session 0x10000325aa60001 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60001 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-16 08:41:24,761] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 08:41:24,762] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-16 08:41:24,762] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-16 08:41:24,763] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,763] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,763] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,764] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:61823, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,767] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60002, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 08:41:24,767] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-16 08:41:24,806] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-16 08:41:24,810] INFO Stat of the created znode at /brokers/ids/0 is: 197,197,1645029684806,1645029684806,1,0,0,72057810307121154,202,0,197
 (kafka.zk.KafkaZkClient)
[2022-02-16 08:41:24,810] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 197 (kafka.zk.KafkaZkClient)
[2022-02-16 12:32:35,269] INFO Expiring session 0x10000325aa60002, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 12:32:35,278] WARN Client session timed out, have not heard from server in 169009ms for session id 0x10000325aa60002 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:35,283] WARN Session 0x10000325aa60002 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 169009ms for session id 0x10000325aa60002
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-16 12:32:36,740] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:36,877] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:36,878] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:51876, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,108] INFO Invalid session 0x10000325aa60002 for client /[0:0:0:0:0:0:0:1]:51876, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 12:32:37,112] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60002 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,112] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 12:32:37,112] INFO EventThread shut down for session: 0x10000325aa60002 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,119] WARN Session 0x10000325aa60002 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60002 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-16 12:32:37,124] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 12:32:37,125] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-16 12:32:37,127] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-16 12:32:37,127] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,128] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-16 12:32:37,172] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,176] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,177] INFO Socket connection established, initiating session, client: /127.0.0.1:51879, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,180] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa60003, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 12:32:37,181] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-16 12:32:37,183] INFO Stat of the created znode at /brokers/ids/0 is: 202,202,1645043557180,1645043557180,1,0,0,72057810307121155,202,0,202
 (kafka.zk.KafkaZkClient)
[2022-02-16 12:32:37,183] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 202 (kafka.zk.KafkaZkClient)
[2022-02-16 14:31:39,690] INFO Expiring session 0x10000325aa60003, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 14:31:40,246] WARN Client session timed out, have not heard from server in 7131205ms for session id 0x10000325aa60003 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:40,252] WARN Session 0x10000325aa60003 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 7131205ms for session id 0x10000325aa60003
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-16 14:31:41,764] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,765] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,765] INFO Socket connection established, initiating session, client: /127.0.0.1:51919, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,766] INFO Invalid session 0x10000325aa60003 for client /127.0.0.1:51919, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-16 14:31:41,766] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60003 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,766] INFO EventThread shut down for session: 0x10000325aa60003 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,766] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 14:31:41,766] WARN Session 0x10000325aa60003 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60003 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-16 14:31:41,767] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-16 14:31:41,767] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-16 14:31:41,767] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-16 14:31:41,768] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,768] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,769] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,769] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:51922, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,771] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-16 14:31:41,772] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60004, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-16 14:31:41,772] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-16 14:31:41,775] INFO Stat of the created znode at /brokers/ids/0 is: 207,207,1645050701772,1645050701772,1,0,0,72057810307121156,202,0,207
 (kafka.zk.KafkaZkClient)
[2022-02-16 14:31:41,775] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 207 (kafka.zk.KafkaZkClient)
[2022-02-17 09:00:08,254] WARN Session 0x10000325aa60004 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
EndOfStreamException: Unable to read additional data from server sessionid 0x10000325aa60004, likely server has closed socket
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-17 09:00:09,740] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-17 09:00:09,744] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-17 09:00:09,754] INFO Socket connection established, initiating session, client: /127.0.0.1:55610, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 09:00:09,761] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa60004, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:17,145] INFO Expiring session 0x10000325aa60004, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-17 14:49:17,295] WARN Client session timed out, have not heard from server in 7557373ms for session id 0x10000325aa60004 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:17,304] WARN Session 0x10000325aa60004 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 7557373ms for session id 0x10000325aa60004
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-17 14:49:19,076] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,077] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,078] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:55664, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,079] INFO Invalid session 0x10000325aa60004 for client /[0:0:0:0:0:0:0:1]:55664, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-17 14:49:19,080] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60004 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,080] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-17 14:49:19,080] WARN Session 0x10000325aa60004 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60004 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-17 14:49:19,080] INFO EventThread shut down for session: 0x10000325aa60004 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,081] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-17 14:49:19,082] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-17 14:49:19,084] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-17 14:49:19,085] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,085] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,086] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,087] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:55667, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,091] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60005, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:49:19,097] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-17 14:49:19,100] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-17 14:49:19,104] INFO Stat of the created znode at /brokers/ids/0 is: 212,212,1645138159101,1645138159101,1,0,0,72057810307121157,202,0,212
 (kafka.zk.KafkaZkClient)
[2022-02-17 14:49:19,104] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 212 (kafka.zk.KafkaZkClient)
[2022-02-17 14:51:06,115] INFO Expiring session 0x10000325aa60005, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-17 14:51:06,284] WARN Client session timed out, have not heard from server in 95297ms for session id 0x10000325aa60005 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:06,285] WARN Session 0x10000325aa60005 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 95297ms for session id 0x10000325aa60005
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-17 14:51:06,287] WARN Unexpected exception (org.apache.zookeeper.server.NIOServerCnxn)
EndOfStreamException: Unable to read additional data from client, it probably closed the socket: address = /[0:0:0:0:0:0:0:1]:55667, session = 0x10000325aa60005
	at org.apache.zookeeper.server.NIOServerCnxn.handleFailedRead(NIOServerCnxn.java:163)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:326)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-17 14:51:07,489] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,514] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,518] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:50294, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,777] INFO Invalid session 0x10000325aa60005 for client /[0:0:0:0:0:0:0:1]:50294, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-17 14:51:07,779] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60005 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,779] INFO EventThread shut down for session: 0x10000325aa60005 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,779] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-17 14:51:07,779] WARN Session 0x10000325aa60005 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60005 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-17 14:51:07,779] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-17 14:51:07,780] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-17 14:51:07,780] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-17 14:51:07,781] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,781] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,781] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,781] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-17 14:51:07,782] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:50299, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,785] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60006, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-17 14:51:07,786] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-17 14:51:07,788] INFO Stat of the created znode at /brokers/ids/0 is: 217,217,1645138267785,1645138267785,1,0,0,72057810307121158,202,0,217
 (kafka.zk.KafkaZkClient)
[2022-02-17 14:51:07,837] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 217 (kafka.zk.KafkaZkClient)
[2022-02-18 07:02:02,770] INFO Expiring session 0x10000325aa60006, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 07:02:03,256] WARN Client session timed out, have not heard from server in 35511845ms for session id 0x10000325aa60006 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:03,271] WARN Session 0x10000325aa60006 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 35511845ms for session id 0x10000325aa60006
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-18 07:02:04,929] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,929] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,931] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:49333, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,933] INFO Invalid session 0x10000325aa60006 for client /[0:0:0:0:0:0:0:1]:49333, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 07:02:04,934] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60006 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,934] INFO EventThread shut down for session: 0x10000325aa60006 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,934] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 07:02:04,934] WARN Session 0x10000325aa60006 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60006 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-18 07:02:04,937] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 07:02:04,937] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-18 07:02:04,941] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-18 07:02:04,942] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,946] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,947] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,947] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:49339, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,953] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60007, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 07:02:04,956] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-18 07:02:04,958] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-18 07:02:04,964] INFO Stat of the created znode at /brokers/ids/0 is: 222,222,1645196524958,1645196524958,1,0,0,72057810307121159,202,0,222
 (kafka.zk.KafkaZkClient)
[2022-02-18 07:02:04,964] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 222 (kafka.zk.KafkaZkClient)
[2022-02-18 10:33:45,772] INFO Expiring session 0x10000325aa60007, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 10:33:45,781] WARN Client session timed out, have not heard from server in 80397ms for session id 0x10000325aa60007 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:45,788] WARN Session 0x10000325aa60007 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 80397ms for session id 0x10000325aa60007
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-18 10:33:47,714] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,714] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,714] INFO Socket connection established, initiating session, client: /127.0.0.1:57287, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,715] INFO Invalid session 0x10000325aa60007 for client /127.0.0.1:57287, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 10:33:47,715] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60007 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,715] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 10:33:47,715] INFO EventThread shut down for session: 0x10000325aa60007 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,716] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 10:33:47,715] WARN Session 0x10000325aa60007 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60007 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-18 10:33:47,716] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-18 10:33:47,717] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-18 10:33:47,717] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,718] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-18 10:33:47,718] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,718] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,718] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:57290, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,721] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60008, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:33:47,722] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-18 10:33:47,723] INFO Stat of the created znode at /brokers/ids/0 is: 227,227,1645209227721,1645209227721,1,0,0,72057810307121160,202,0,227
 (kafka.zk.KafkaZkClient)
[2022-02-18 10:33:47,723] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 227 (kafka.zk.KafkaZkClient)
[2022-02-18 10:47:23,429] WARN Client session timed out, have not heard from server in 16917ms for session id 0x10000325aa60008 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:47:23,431] WARN Session 0x10000325aa60008 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 16917ms for session id 0x10000325aa60008
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-18 10:47:23,435] WARN Unexpected exception (org.apache.zookeeper.server.NIOServerCnxn)
EndOfStreamException: Unable to read additional data from client, it probably closed the socket: address = /[0:0:0:0:0:0:0:1]:57290, session = 0x10000325aa60008
	at org.apache.zookeeper.server.NIOServerCnxn.handleFailedRead(NIOServerCnxn.java:163)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:326)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-18 10:47:24,592] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:47:24,592] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:47:24,593] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:64936, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 10:47:24,594] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60008, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:18,865] WARN Client session timed out, have not heard from server in 17685590ms for session id 0x10000325aa60008 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:19,512] WARN Session 0x10000325aa60008 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 17685590ms for session id 0x10000325aa60008
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-18 16:21:19,517] WARN Unexpected exception (org.apache.zookeeper.server.NIOServerCnxn)
EndOfStreamException: Unable to read additional data from client, it probably closed the socket: address = /[0:0:0:0:0:0:0:1]:64936, session = 0x10000325aa60008
	at org.apache.zookeeper.server.NIOServerCnxn.handleFailedRead(NIOServerCnxn.java:163)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:326)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-18 16:21:19,663] INFO Expiring session 0x10000325aa60008, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 16:21:20,730] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,800] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,803] INFO Socket connection established, initiating session, client: /127.0.0.1:50833, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,967] INFO Invalid session 0x10000325aa60008 for client /127.0.0.1:50833, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 16:21:20,971] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60008 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,971] INFO EventThread shut down for session: 0x10000325aa60008 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,971] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 16:21:20,977] WARN Session 0x10000325aa60008 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60008 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-18 16:21:20,977] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 16:21:20,977] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-18 16:21:20,978] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-18 16:21:20,978] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,979] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,979] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,979] INFO Socket connection established, initiating session, client: /127.0.0.1:54935, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:20,981] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-18 16:21:20,982] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa60009, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 16:21:21,040] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-18 16:21:21,043] INFO Stat of the created znode at /brokers/ids/0 is: 232,232,1645230081039,1645230081039,1,0,0,72057810307121161,202,0,232
 (kafka.zk.KafkaZkClient)
[2022-02-18 16:21:21,043] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 232 (kafka.zk.KafkaZkClient)
[2022-02-18 20:09:26,119] WARN Client session timed out, have not heard from server in 13601095ms for session id 0x10000325aa60009 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:26,485] WARN Session 0x10000325aa60009 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 13601095ms for session id 0x10000325aa60009
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-18 20:09:26,494] WARN Unexpected exception (org.apache.zookeeper.server.NIOServerCnxn)
EndOfStreamException: Unable to read additional data from client, it probably closed the socket: address = /127.0.0.1:54935, session = 0x10000325aa60009
	at org.apache.zookeeper.server.NIOServerCnxn.handleFailedRead(NIOServerCnxn.java:163)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:326)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-18 20:09:26,967] INFO Expiring session 0x10000325aa60009, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 20:09:27,958] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,260] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,483] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:64164, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,489] INFO Invalid session 0x10000325aa60009 for client /[0:0:0:0:0:0:0:1]:64164, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-18 20:09:28,493] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60009 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,493] INFO EventThread shut down for session: 0x10000325aa60009 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,493] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 20:09:28,497] WARN Session 0x10000325aa60009 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60009 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-18 20:09:28,497] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-18 20:09:28,498] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-18 20:09:28,499] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-18 20:09:28,499] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,500] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,500] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,500] INFO Socket connection established, initiating session, client: /127.0.0.1:64167, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,503] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-18 20:09:28,503] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa6000a, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-18 20:09:28,582] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-18 20:09:28,585] INFO Stat of the created znode at /brokers/ids/0 is: 237,237,1645243768581,1645243768581,1,0,0,72057810307121162,202,0,237
 (kafka.zk.KafkaZkClient)
[2022-02-18 20:09:28,650] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 237 (kafka.zk.KafkaZkClient)
[2022-02-19 10:24:30,786] INFO Expiring session 0x10000325aa6000a, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-19 10:24:31,278] WARN Client session timed out, have not heard from server in 44484342ms for session id 0x10000325aa6000a (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:31,293] WARN Session 0x10000325aa6000a for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 44484342ms for session id 0x10000325aa6000a
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-19 10:24:33,096] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,097] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,098] INFO Socket connection established, initiating session, client: /127.0.0.1:51502, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,098] INFO Invalid session 0x10000325aa6000a for client /127.0.0.1:51502, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-19 10:24:33,099] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000a has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,099] WARN Session 0x10000325aa6000a for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000a has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-19 10:24:33,099] INFO EventThread shut down for session: 0x10000325aa6000a (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,099] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-19 10:24:33,100] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-19 10:24:33,101] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-19 10:24:33,102] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-19 10:24:33,102] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,103] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,103] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,103] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:51505, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,107] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-19 10:24:33,107] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa6000b, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 10:24:33,108] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-19 10:24:33,110] INFO Stat of the created znode at /brokers/ids/0 is: 242,242,1645295073108,1645295073108,1,0,0,72057810307121163,202,0,242
 (kafka.zk.KafkaZkClient)
[2022-02-19 10:24:33,110] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 242 (kafka.zk.KafkaZkClient)
[2022-02-19 20:48:05,703] WARN Client session timed out, have not heard from server in 1500301ms for session id 0x10000325aa6000b (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:06,294] WARN Session 0x10000325aa6000b for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 1500301ms for session id 0x10000325aa6000b
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-19 20:48:06,751] INFO Expiring session 0x10000325aa6000b, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-19 20:48:07,444] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,474] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,493] INFO Socket connection established, initiating session, client: /127.0.0.1:49911, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,608] INFO Invalid session 0x10000325aa6000b for client /127.0.0.1:49911, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-19 20:48:07,609] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000b has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,609] INFO EventThread shut down for session: 0x10000325aa6000b (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,609] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-19 20:48:07,609] WARN Session 0x10000325aa6000b for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000b has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-19 20:48:07,610] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-19 20:48:07,610] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-19 20:48:07,611] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-19 20:48:07,611] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,622] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-19 20:48:07,630] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,631] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,634] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:49914, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,641] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa6000c, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-19 20:48:07,642] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-19 20:48:07,648] INFO Stat of the created znode at /brokers/ids/0 is: 247,247,1645332487641,1645332487641,1,0,0,72057810307121164,202,0,247
 (kafka.zk.KafkaZkClient)
[2022-02-19 20:48:07,648] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 247 (kafka.zk.KafkaZkClient)
[2022-02-20 09:37:36,814] WARN Client session timed out, have not heard from server in 41243102ms for session id 0x10000325aa6000c (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:36,820] WARN Session 0x10000325aa6000c for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 41243102ms for session id 0x10000325aa6000c
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-20 09:37:36,821] WARN Unexpected exception (org.apache.zookeeper.server.NIOServerCnxn)
EndOfStreamException: Unable to read additional data from client, it probably closed the socket: address = /[0:0:0:0:0:0:0:1]:49914, session = 0x10000325aa6000c
	at org.apache.zookeeper.server.NIOServerCnxn.handleFailedRead(NIOServerCnxn.java:163)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:326)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-20 09:37:38,161] INFO Expiring session 0x10000325aa6000c, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 09:37:38,357] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,357] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,358] INFO Socket connection established, initiating session, client: /127.0.0.1:61008, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,359] INFO Invalid session 0x10000325aa6000c for client /127.0.0.1:61008, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 09:37:38,359] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000c has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,359] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 09:37:38,359] INFO EventThread shut down for session: 0x10000325aa6000c (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,360] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 09:37:38,359] WARN Session 0x10000325aa6000c for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000c has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-20 09:37:38,360] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-20 09:37:38,361] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-20 09:37:38,361] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,362] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,362] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,362] INFO Socket connection established, initiating session, client: /127.0.0.1:61011, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,365] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-20 09:37:38,373] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa6000d, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 09:37:38,373] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-20 09:37:38,385] INFO Stat of the created znode at /brokers/ids/0 is: 252,252,1645378658373,1645378658373,1,0,0,72057810307121165,202,0,252
 (kafka.zk.KafkaZkClient)
[2022-02-20 09:37:38,385] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 252 (kafka.zk.KafkaZkClient)
[2022-02-20 10:48:19,022] INFO Expiring session 0x10000325aa6000d, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 10:48:19,038] WARN Client session timed out, have not heard from server in 3399726ms for session id 0x10000325aa6000d (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:19,047] WARN Session 0x10000325aa6000d for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 3399726ms for session id 0x10000325aa6000d
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-20 10:48:20,936] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,936] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,937] INFO Socket connection established, initiating session, client: /127.0.0.1:59464, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,938] INFO Invalid session 0x10000325aa6000d for client /127.0.0.1:59464, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 10:48:20,938] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000d has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,938] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 10:48:20,938] INFO EventThread shut down for session: 0x10000325aa6000d (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,938] WARN Session 0x10000325aa6000d for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000d has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-20 10:48:20,939] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 10:48:20,939] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-20 10:48:20,940] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-20 10:48:20,940] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,941] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,942] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,943] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:59467, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,944] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-20 10:48:20,947] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa6000e, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:48:20,948] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-20 10:48:20,950] INFO Stat of the created znode at /brokers/ids/0 is: 257,257,1645382900948,1645382900948,1,0,0,72057810307121166,202,0,257
 (kafka.zk.KafkaZkClient)
[2022-02-20 10:48:20,950] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 257 (kafka.zk.KafkaZkClient)
[2022-02-20 10:49:26,857] INFO Expiring session 0x10000325aa6000e, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 10:49:26,859] WARN Client session timed out, have not heard from server in 18031ms for session id 0x10000325aa6000e (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:26,860] WARN Session 0x10000325aa6000e for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 18031ms for session id 0x10000325aa6000e
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-20 10:49:28,111] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,185] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,193] INFO Socket connection established, initiating session, client: /127.0.0.1:53826, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,193] INFO Invalid session 0x10000325aa6000e for client /127.0.0.1:53826, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 10:49:28,195] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000e has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,195] INFO EventThread shut down for session: 0x10000325aa6000e (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,195] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 10:49:28,265] WARN Session 0x10000325aa6000e for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000e has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-20 10:49:28,366] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 10:49:28,371] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-20 10:49:28,378] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-20 10:49:28,447] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,501] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-20 10:49:28,509] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,509] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,509] INFO Socket connection established, initiating session, client: /127.0.0.1:53829, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,512] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa6000f, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 10:49:28,513] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-20 10:49:28,514] INFO Stat of the created znode at /brokers/ids/0 is: 262,262,1645382968512,1645382968512,1,0,0,72057810307121167,202,0,262
 (kafka.zk.KafkaZkClient)
[2022-02-20 10:49:28,514] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 262 (kafka.zk.KafkaZkClient)
[2022-02-20 16:16:18,058] INFO Expiring session 0x10000325aa6000f, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 16:16:18,081] WARN Client session timed out, have not heard from server in 16113403ms for session id 0x10000325aa6000f (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:18,088] WARN Session 0x10000325aa6000f for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 16113403ms for session id 0x10000325aa6000f
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-20 16:16:19,782] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:19,875] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:19,885] INFO Socket connection established, initiating session, client: /127.0.0.1:54709, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,112] INFO Invalid session 0x10000325aa6000f for client /127.0.0.1:54709, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 16:16:20,118] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa6000f has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,118] INFO EventThread shut down for session: 0x10000325aa6000f (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,118] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 16:16:20,128] WARN Session 0x10000325aa6000f for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa6000f has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-20 16:16:20,129] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 16:16:20,129] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-20 16:16:20,130] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-20 16:16:20,130] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,133] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,135] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-20 16:16:20,178] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,180] INFO Socket connection established, initiating session, client: /127.0.0.1:54713, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,182] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa60010, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 16:16:20,184] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-20 16:16:20,187] INFO Stat of the created znode at /brokers/ids/0 is: 267,267,1645402580183,1645402580183,1,0,0,72057810307121168,202,0,267
 (kafka.zk.KafkaZkClient)
[2022-02-20 16:16:20,187] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 267 (kafka.zk.KafkaZkClient)
[2022-02-20 17:24:28,033] INFO Expiring session 0x10000325aa60010, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 17:24:28,042] WARN Client session timed out, have not heard from server in 27141ms for session id 0x10000325aa60010 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:28,048] WARN Session 0x10000325aa60010 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 27141ms for session id 0x10000325aa60010
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-20 17:24:29,820] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:29,884] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:29,892] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:54163, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:29,893] INFO Invalid session 0x10000325aa60010 for client /[0:0:0:0:0:0:0:1]:54163, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-20 17:24:29,903] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60010 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:29,903] INFO EventThread shut down for session: 0x10000325aa60010 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:29,903] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 17:24:29,969] WARN Session 0x10000325aa60010 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60010 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-20 17:24:29,973] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-20 17:24:29,973] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-20 17:24:29,974] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-20 17:24:30,000] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:30,001] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:30,001] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:30,001] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-20 17:24:30,002] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:54166, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:30,006] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60011, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-20 17:24:30,007] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-20 17:24:30,009] INFO Stat of the created znode at /brokers/ids/0 is: 272,272,1645406670006,1645406670006,1,0,0,72057810307121169,202,0,272
 (kafka.zk.KafkaZkClient)
[2022-02-20 17:24:30,017] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 272 (kafka.zk.KafkaZkClient)
[2022-02-21 06:02:48,149] INFO Expiring session 0x10000325aa60011, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 06:02:49,498] WARN Client session timed out, have not heard from server in 38260698ms for session id 0x10000325aa60011 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:49,633] WARN Session 0x10000325aa60011 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 38260698ms for session id 0x10000325aa60011
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 06:02:51,039] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:51,923] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:51,941] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:57078, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,187] INFO Invalid session 0x10000325aa60011 for client /[0:0:0:0:0:0:0:1]:57078, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 06:02:52,193] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60011 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,206] WARN Session 0x10000325aa60011 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60011 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 06:02:52,194] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 06:02:52,193] INFO EventThread shut down for session: 0x10000325aa60011 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,206] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 06:02:52,207] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 06:02:52,210] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 06:02:52,264] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,322] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 06:02:52,336] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,347] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,348] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:57082, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,359] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60012, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 06:02:52,363] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 06:02:52,365] INFO Stat of the created znode at /brokers/ids/0 is: 277,277,1645452172362,1645452172362,1,0,0,72057810307121170,202,0,277
 (kafka.zk.KafkaZkClient)
[2022-02-21 06:02:52,389] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 277 (kafka.zk.KafkaZkClient)
[2022-02-21 10:09:27,101] INFO Expiring session 0x10000325aa60012, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 10:09:27,122] WARN Client session timed out, have not heard from server in 7573787ms for session id 0x10000325aa60012 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:27,128] WARN Session 0x10000325aa60012 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 7573787ms for session id 0x10000325aa60012
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 10:09:28,971] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,971] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,973] INFO Socket connection established, initiating session, client: /127.0.0.1:49576, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,974] INFO Invalid session 0x10000325aa60012 for client /127.0.0.1:49576, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 10:09:28,975] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60012 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,975] WARN Session 0x10000325aa60012 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60012 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 10:09:28,976] INFO EventThread shut down for session: 0x10000325aa60012 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,976] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 10:09:28,978] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 10:09:28,978] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 10:09:28,980] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 10:09:28,980] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,981] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,981] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,982] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:49579, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,983] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 10:09:28,989] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60013, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 10:09:28,990] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 10:09:28,992] INFO Stat of the created znode at /brokers/ids/0 is: 282,282,1645466968990,1645466968990,1,0,0,72057810307121171,202,0,282
 (kafka.zk.KafkaZkClient)
[2022-02-21 10:09:28,993] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 282 (kafka.zk.KafkaZkClient)
[2022-02-21 14:03:24,875] INFO Expiring session 0x10000325aa60013, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 14:03:25,244] WARN Client session timed out, have not heard from server in 4009944ms for session id 0x10000325aa60013 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:25,258] WARN Session 0x10000325aa60013 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 4009944ms for session id 0x10000325aa60013
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 14:03:27,193] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,193] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,195] INFO Socket connection established, initiating session, client: /127.0.0.1:54162, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,196] INFO Invalid session 0x10000325aa60013 for client /127.0.0.1:54162, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 14:03:27,196] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60013 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,196] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 14:03:27,196] INFO EventThread shut down for session: 0x10000325aa60013 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,198] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 14:03:27,196] WARN Session 0x10000325aa60013 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60013 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 14:03:27,198] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 14:03:27,199] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 14:03:27,199] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,200] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,200] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,201] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:54165, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,202] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 14:03:27,208] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60014, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 14:03:27,209] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 14:03:27,216] INFO Stat of the created znode at /brokers/ids/0 is: 287,287,1645481007209,1645481007209,1,0,0,72057810307121172,202,0,287
 (kafka.zk.KafkaZkClient)
[2022-02-21 14:03:27,217] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 287 (kafka.zk.KafkaZkClient)
[2022-02-21 16:30:22,113] INFO Expiring session 0x10000325aa60014, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 16:30:22,304] WARN Client session timed out, have not heard from server in 8322635ms for session id 0x10000325aa60014 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:22,311] WARN Session 0x10000325aa60014 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 8322635ms for session id 0x10000325aa60014
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 16:30:23,443] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,443] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,447] INFO Socket connection established, initiating session, client: /127.0.0.1:50197, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,512] INFO Invalid session 0x10000325aa60014 for client /127.0.0.1:50197, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 16:30:23,513] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60014 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,513] INFO EventThread shut down for session: 0x10000325aa60014 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,513] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 16:30:23,522] WARN Session 0x10000325aa60014 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60014 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 16:30:23,527] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 16:30:23,537] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 16:30:23,540] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 16:30:23,543] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,548] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 16:30:23,549] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,550] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,551] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:50200, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,557] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60015, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:30:23,557] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 16:30:23,562] INFO Stat of the created znode at /brokers/ids/0 is: 292,292,1645489823557,1645489823557,1,0,0,72057810307121173,202,0,292
 (kafka.zk.KafkaZkClient)
[2022-02-21 16:30:23,562] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 292 (kafka.zk.KafkaZkClient)
[2022-02-21 16:59:38,554] INFO Expiring session 0x10000325aa60015, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 16:59:39,203] WARN Client session timed out, have not heard from server in 1539543ms for session id 0x10000325aa60015 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:39,209] WARN Session 0x10000325aa60015 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 1539543ms for session id 0x10000325aa60015
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 16:59:40,524] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,524] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,525] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:54864, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,526] INFO Invalid session 0x10000325aa60015 for client /[0:0:0:0:0:0:0:1]:54864, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 16:59:40,526] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60015 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,526] INFO EventThread shut down for session: 0x10000325aa60015 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,526] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 16:59:40,526] WARN Session 0x10000325aa60015 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60015 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 16:59:40,527] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 16:59:40,528] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 16:59:40,528] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 16:59:40,528] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,529] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 16:59:40,529] INFO Opening socket connection to server localhost/127.0.0.1:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,529] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,529] INFO Socket connection established, initiating session, client: /127.0.0.1:54867, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,533] INFO Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000325aa60016, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 16:59:40,534] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 16:59:40,536] INFO Stat of the created znode at /brokers/ids/0 is: 297,297,1645491580533,1645491580533,1,0,0,72057810307121174,202,0,297
 (kafka.zk.KafkaZkClient)
[2022-02-21 16:59:40,536] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 297 (kafka.zk.KafkaZkClient)
[2022-02-21 19:12:31,265] INFO Expiring session 0x10000325aa60016, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 19:12:31,285] WARN Client session timed out, have not heard from server in 5898282ms for session id 0x10000325aa60016 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:31,293] WARN Session 0x10000325aa60016 for sever localhost/127.0.0.1:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionTimeoutException: Client session timed out, have not heard from server in 5898282ms for session id 0x10000325aa60016
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1258)
[2022-02-21 19:12:33,343] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,343] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,344] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:59642, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,344] INFO Invalid session 0x10000325aa60016 for client /[0:0:0:0:0:0:0:1]:59642, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2022-02-21 19:12:33,344] WARN Unable to reconnect to ZooKeeper service, session 0x10000325aa60016 has expired (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,345] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 19:12:33,345] INFO EventThread shut down for session: 0x10000325aa60016 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,345] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2022-02-21 19:12:33,345] WARN Session 0x10000325aa60016 for sever localhost/[0:0:0:0:0:0:0:1]:2181, Closing socket connection. Attempting reconnect except it is a SessionExpiredException. (org.apache.zookeeper.ClientCnxn)
org.apache.zookeeper.ClientCnxn$SessionExpiredException: Unable to reconnect to ZooKeeper service, session 0x10000325aa60016 has expired
	at org.apache.zookeeper.ClientCnxn$SendThread.onConnected(ClientCnxn.java:1434)
	at org.apache.zookeeper.ClientCnxnSocket.readConnectResult(ClientCnxnSocket.java:154)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:86)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290)
[2022-02-21 19:12:33,346] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@14fa86ae (org.apache.zookeeper.ZooKeeper)
[2022-02-21 19:12:33,346] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2022-02-21 19:12:33,346] INFO zookeeper.request.timeout value is 0. feature enabled=false (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,347] INFO Opening socket connection to server localhost/[0:0:0:0:0:0:0:1]:2181. (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,347] INFO SASL config status: Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,347] INFO Socket connection established, initiating session, client: /[0:0:0:0:0:0:0:1]:59645, server: localhost/[0:0:0:0:0:0:0:1]:2181 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,349] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2022-02-21 19:12:33,354] INFO Session establishment complete on server localhost/[0:0:0:0:0:0:0:1]:2181, session id = 0x10000325aa60017, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2022-02-21 19:12:33,355] INFO Updated cache from existing FinalizedFeaturesAndEpoch(features=Features{}, epoch=0) to latest FinalizedFeaturesAndEpoch(features=Features{}, epoch=0). (kafka.server.FinalizedFeatureCache)
[2022-02-21 19:12:33,365] INFO Stat of the created znode at /brokers/ids/0 is: 302,302,1645499553355,1645499553355,1,0,0,72057810307121175,202,0,302
 (kafka.zk.KafkaZkClient)
[2022-02-21 19:12:33,365] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://AndrewsPC:9092, czxid (broker epoch): 302 (kafka.zk.KafkaZkClient)
[2022-02-22 10:10:01,634] INFO [LocalLog partition=normal-0, dir=c:\kafkatwo\kafka-logs2] Rolled new log segment at offset 369109 in 28 ms. (kafka.log.LocalLog)
[2022-02-22 10:10:01,646] INFO [ProducerStateManager partition=normal-0] Wrote producer snapshot at offset 369109 with 0 producer ids in 5 ms. (kafka.log.ProducerStateManager)
[2022-02-22 10:10:01,648] INFO [UnifiedLog partition=normal-0, dir=c:\kafkatwo\kafka-logs2] Deleting segment LogSegment(baseOffset=0, size=56960225, lastModifiedTime=1644948302192, largestRecordTimestamp=Some(1644948302191)) due to retention time 604800000ms breach based on the largest record timestamp in the segment (kafka.log.UnifiedLog)
[2022-02-22 10:10:01,653] ERROR Error while deleting segments for normal-0 in dir c:\kafkatwo\kafka-logs2 (kafka.server.LogDirFailureChannel)
java.nio.file.FileSystemException: c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index -> c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index.deleted: The process cannot access the file because it is being used by another process
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:395)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:292)
	at java.base/java.nio.file.Files.move(Files.java:1426)
	at org.apache.kafka.common.utils.Utils.atomicMoveWithFallback(Utils.java:934)
	at kafka.log.AbstractIndex.renameTo(AbstractIndex.scala:210)
	at kafka.log.LazyIndex$IndexValue.renameTo(LazyIndex.scala:155)
	at kafka.log.LazyIndex.$anonfun$renameTo$1(LazyIndex.scala:79)
	at kafka.log.LazyIndex.renameTo(LazyIndex.scala:79)
	at kafka.log.LogSegment.changeFileSuffixes(LogSegment.scala:492)
	at kafka.log.LocalLog$.$anonfun$deleteSegmentFiles$1(LocalLog.scala:940)
	at kafka.log.LocalLog$.$anonfun$deleteSegmentFiles$1$adapted(LocalLog.scala:940)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at kafka.log.LocalLog$.deleteSegmentFiles(LocalLog.scala:940)
	at kafka.log.LocalLog.removeAndDeleteSegments(LocalLog.scala:310)
	at kafka.log.UnifiedLog.$anonfun$deleteSegments$2(UnifiedLog.scala:1358)
	at kafka.log.UnifiedLog.deleteSegments(UnifiedLog.scala:1707)
	at kafka.log.UnifiedLog.deleteRetentionMsBreachedSegments(UnifiedLog.scala:1342)
	at kafka.log.UnifiedLog.deleteOldSegments(UnifiedLog.scala:1377)
	at kafka.log.LogManager.$anonfun$cleanupLogs$3(LogManager.scala:1169)
	at kafka.log.LogManager.$anonfun$cleanupLogs$3$adapted(LogManager.scala:1166)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at kafka.log.LogManager.cleanupLogs(LogManager.scala:1166)
	at kafka.log.LogManager.$anonfun$startupWithConfigOverrides$2(LogManager.scala:462)
	at kafka.utils.KafkaScheduler.$anonfun$schedule$2(KafkaScheduler.scala:116)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
	Suppressed: java.nio.file.FileSystemException: c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index -> c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index.deleted: The process cannot access the file because it is being used by another process
		at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
		at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:309)
		at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:292)
		at java.base/java.nio.file.Files.move(Files.java:1426)
		at org.apache.kafka.common.utils.Utils.atomicMoveWithFallback(Utils.java:931)
		... 26 more
[2022-02-22 10:10:01,657] WARN [ReplicaManager broker=0] Stopping serving replicas in dir c:\kafkatwo\kafka-logs2 (kafka.server.ReplicaManager)
[2022-02-22 10:10:01,658] ERROR Uncaught exception in scheduled task 'kafka-log-retention' (kafka.utils.KafkaScheduler)
org.apache.kafka.common.errors.KafkaStorageException: Error while deleting segments for normal-0 in dir c:\kafkatwo\kafka-logs2
Caused by: java.nio.file.FileSystemException: c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index -> c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index.deleted: The process cannot access the file because it is being used by another process
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:395)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:292)
	at java.base/java.nio.file.Files.move(Files.java:1426)
	at org.apache.kafka.common.utils.Utils.atomicMoveWithFallback(Utils.java:934)
	at kafka.log.AbstractIndex.renameTo(AbstractIndex.scala:210)
	at kafka.log.LazyIndex$IndexValue.renameTo(LazyIndex.scala:155)
	at kafka.log.LazyIndex.$anonfun$renameTo$1(LazyIndex.scala:79)
	at kafka.log.LazyIndex.renameTo(LazyIndex.scala:79)
	at kafka.log.LogSegment.changeFileSuffixes(LogSegment.scala:492)
	at kafka.log.LocalLog$.$anonfun$deleteSegmentFiles$1(LocalLog.scala:940)
	at kafka.log.LocalLog$.$anonfun$deleteSegmentFiles$1$adapted(LocalLog.scala:940)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at kafka.log.LocalLog$.deleteSegmentFiles(LocalLog.scala:940)
	at kafka.log.LocalLog.removeAndDeleteSegments(LocalLog.scala:310)
	at kafka.log.UnifiedLog.$anonfun$deleteSegments$2(UnifiedLog.scala:1358)
	at kafka.log.UnifiedLog.deleteSegments(UnifiedLog.scala:1707)
	at kafka.log.UnifiedLog.deleteRetentionMsBreachedSegments(UnifiedLog.scala:1342)
	at kafka.log.UnifiedLog.deleteOldSegments(UnifiedLog.scala:1377)
	at kafka.log.LogManager.$anonfun$cleanupLogs$3(LogManager.scala:1169)
	at kafka.log.LogManager.$anonfun$cleanupLogs$3$adapted(LogManager.scala:1166)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at kafka.log.LogManager.cleanupLogs(LogManager.scala:1166)
	at kafka.log.LogManager.$anonfun$startupWithConfigOverrides$2(LogManager.scala:462)
	at kafka.utils.KafkaScheduler.$anonfun$schedule$2(KafkaScheduler.scala:116)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
	Suppressed: java.nio.file.FileSystemException: c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index -> c:\kafkatwo\kafka-logs2\normal-0\00000000000000000000.index.deleted: The process cannot access the file because it is being used by another process
		at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
		at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:309)
		at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:292)
		at java.base/java.nio.file.Files.move(Files.java:1426)
		at org.apache.kafka.common.utils.Utils.atomicMoveWithFallback(Utils.java:931)
		... 26 more
[2022-02-22 10:10:01,658] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions HashSet(__consumer_offsets-22, __consumer_offsets-25, __consumer_offsets-35, raw-0, __consumer_offsets-37, __consumer_offsets-38, __consumer_offsets-30, normal-0, __consumer_offsets-8, __consumer_offsets-21, __consumer_offsets-4, __consumer_offsets-27, __consumer_offsets-7, __consumer_offsets-9, __consumer_offsets-46, __consumer_offsets-41, __consumer_offsets-33, __consumer_offsets-23, __consumer_offsets-49, __consumer_offsets-47, __consumer_offsets-16, __consumer_offsets-28, __consumer_offsets-31, __consumer_offsets-36, __consumer_offsets-42, __consumer_offsets-3, __consumer_offsets-18, __consumer_offsets-15, __consumer_offsets-24, __consumer_offsets-17, __consumer_offsets-48, __consumer_offsets-19, __consumer_offsets-11, ingest-0, __consumer_offsets-13, __consumer_offsets-2, __consumer_offsets-43, __consumer_offsets-6, __consumer_offsets-14, __consumer_offsets-20, __consumer_offsets-0, __consumer_offsets-44, __consumer_offsets-39, __consumer_offsets-12, __consumer_offsets-45, __consumer_offsets-1, __consumer_offsets-5, __consumer_offsets-26, __consumer_offsets-29, __consumer_offsets-34, __consumer_offsets-10, __consumer_offsets-32, __consumer_offsets-40) (kafka.server.ReplicaFetcherManager)
[2022-02-22 10:10:01,659] INFO [ReplicaAlterLogDirsManager on broker 0] Removed fetcher for partitions HashSet(__consumer_offsets-22, __consumer_offsets-25, __consumer_offsets-35, raw-0, __consumer_offsets-37, __consumer_offsets-38, __consumer_offsets-30, normal-0, __consumer_offsets-8, __consumer_offsets-21, __consumer_offsets-4, __consumer_offsets-27, __consumer_offsets-7, __consumer_offsets-9, __consumer_offsets-46, __consumer_offsets-41, __consumer_offsets-33, __consumer_offsets-23, __consumer_offsets-49, __consumer_offsets-47, __consumer_offsets-16, __consumer_offsets-28, __consumer_offsets-31, __consumer_offsets-36, __consumer_offsets-42, __consumer_offsets-3, __consumer_offsets-18, __consumer_offsets-15, __consumer_offsets-24, __consumer_offsets-17, __consumer_offsets-48, __consumer_offsets-19, __consumer_offsets-11, ingest-0, __consumer_offsets-13, __consumer_offsets-2, __consumer_offsets-43, __consumer_offsets-6, __consumer_offsets-14, __consumer_offsets-20, __consumer_offsets-0, __consumer_offsets-44, __consumer_offsets-39, __consumer_offsets-12, __consumer_offsets-45, __consumer_offsets-1, __consumer_offsets-5, __consumer_offsets-26, __consumer_offsets-29, __consumer_offsets-34, __consumer_offsets-10, __consumer_offsets-32, __consumer_offsets-40) (kafka.server.ReplicaAlterLogDirsManager)
[2022-02-22 10:10:01,670] WARN [ReplicaManager broker=0] Broker 0 stopped fetcher for partitions __consumer_offsets-22,__consumer_offsets-25,__consumer_offsets-35,raw-0,__consumer_offsets-37,__consumer_offsets-38,__consumer_offsets-30,normal-0,__consumer_offsets-8,__consumer_offsets-21,__consumer_offsets-4,__consumer_offsets-27,__consumer_offsets-7,__consumer_offsets-9,__consumer_offsets-46,__consumer_offsets-41,__consumer_offsets-33,__consumer_offsets-23,__consumer_offsets-49,__consumer_offsets-47,__consumer_offsets-16,__consumer_offsets-28,__consumer_offsets-31,__consumer_offsets-36,__consumer_offsets-42,__consumer_offsets-3,__consumer_offsets-18,__consumer_offsets-15,__consumer_offsets-24,__consumer_offsets-17,__consumer_offsets-48,__consumer_offsets-19,__consumer_offsets-11,ingest-0,__consumer_offsets-13,__consumer_offsets-2,__consumer_offsets-43,__consumer_offsets-6,__consumer_offsets-14,__consumer_offsets-20,__consumer_offsets-0,__consumer_offsets-44,__consumer_offsets-39,__consumer_offsets-12,__consumer_offsets-45,__consumer_offsets-1,__consumer_offsets-5,__consumer_offsets-26,__consumer_offsets-29,__consumer_offsets-34,__consumer_offsets-10,__consumer_offsets-32,__consumer_offsets-40 and stopped moving logs for partitions  because they are in the failed log directory c:\kafkatwo\kafka-logs2. (kafka.server.ReplicaManager)
[2022-02-22 10:10:01,670] WARN Stopping serving logs in dir c:\kafkatwo\kafka-logs2 (kafka.log.LogManager)
[2022-02-22 10:10:01,675] ERROR Shutdown broker because all log dirs in c:\kafkatwo\kafka-logs2 have failed (kafka.log.LogManager)
[2022-02-22 10:10:02,183] WARN Close of session 0x10000325aa60017 (org.apache.zookeeper.server.NIOServerCnxn)
java.net.SocketException: Connection reset
	at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:367)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:398)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:324)
	at org.apache.zookeeper.server.NIOServerCnxnFactory$IOWorkRequest.doWork(NIOServerCnxnFactory.java:522)
	at org.apache.zookeeper.server.WorkerService$ScheduledWorkRequest.run(WorkerService.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
[2022-02-22 10:10:19,237] INFO Expiring session 0x10000325aa60017, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
